query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/Params.java

```diff
@@ -56,6 +56,58 @@ private Params(
     return bytesStringParams;
   }
 
+  @Override
+  public String toString() {
+    return "Params{" +
+        "integerParams=" + integerParams +
+        ", longParams=" + longParams +
+        ", stringParams=" + stringParams +
+        ", floatParams=" + floatParams +
+        ", doubleParams=" + doubleParams +
+        ", bytesStringParams=" + bytesStringParams +
+        '}';
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    Params params = (Params) o;
+
+    if (!integerParams.equals(params.integerParams)) {
+      return false;
+    }
+    if (!longParams.equals(params.longParams)) {
+      return false;
+    }
+    if (!stringParams.equals(params.stringParams)) {
+      return false;
+    }
+    if (!floatParams.equals(params.floatParams)) {
+      return false;
+    }
+    if (!doubleParams.equals(params.doubleParams)) {
+      return false;
+    }
+    return bytesStringParams.equals(params.bytesStringParams);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = integerParams.hashCode();
```

Comment:
Why are you not using Objects methods?

Reply:
Any advantages that gives specifically? I can check that.